pkg/operator/staticpod_interface.go

```diff
@@ -0,0 +1,51 @@
+package operator
+
+import (
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorconfigclientv1alpha1 "github.com/openshift/cluster-kube-apiserver-operator/pkg/generated/clientset/versioned/typed/kubeapiserver/v1alpha1"
+	operatorclientinformers "github.com/openshift/cluster-kube-apiserver-operator/pkg/generated/informers/externalversions"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/staticpodcontroller"
+	"k8s.io/client-go/tools/cache"
+)
+
+type staticPodOperatorClient struct {
+	informers operatorclientinformers.SharedInformerFactory
+	client    operatorconfigclientv1alpha1.KubeapiserverV1alpha1Interface
+}
+
+func (c *staticPodOperatorClient) Informer() cache.SharedIndexInformer {
+	return c.informers.Kubeapiserver().V1alpha1().KubeAPIServerOperatorConfigs().Informer()
+}
+
+func (c *staticPodOperatorClient) Get() (*operatorv1alpha1.OperatorSpec, *staticpodcontroller.StaticPodConfigStatus, string, error) {
+	instance, err := c.informers.Kubeapiserver().V1alpha1().KubeAPIServerOperatorConfigs().Lister().Get("instance")
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	return &instance.Spec.OperatorSpec, &instance.Status.StaticPodConfigStatus, instance.ResourceVersion, nil
+}
+
+func (c *staticPodOperatorClient) UpdateStatus(resourceVersion string, status *staticpodcontroller.StaticPodConfigStatus) (*staticpodcontroller.StaticPodConfigStatus, error) {
+	original, err := c.informers.Kubeapiserver().V1alpha1().KubeAPIServerOperatorConfigs().Lister().Get("instance")
```

Comment:
You're swallowing the error, why not using `CurrentStatus` here also?

Reply:
> You're swallowing the error, why not using `CurrentStatus` here also?

you need a full object of some kind to build the update payload until we switch to a dynamic client.